src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -150,6 +155,9 @@
 
     private static final String API_COMMIT_STATUS_PATH = "/rest/build-status/1.0/commits{/hash}";
     private static final Integer DEFAULT_PAGE_LIMIT = 200;
+    private static final int API_RATE_LIMIT_STATUS_CODE = 429;
+    private static final Duration API_RATE_LIMIT_INITIAL_SLEEP = Main.isUnitTest ? Duration.ofMillis(100) : Duration.ofSeconds(5);
+    private static final Duration API_RATE_LIMIT_MAX_SLEEP = Duration.ofMinutes(30);
```

Comment:
Not sure about the best values here, see https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/581/#discussion_r840053080 for more details.